### PR TITLE
Switch stateful tests to API 2.0

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: src/test/stateful_test_utils.hpp [hunk 1]
```diff
@@ -120,6 +142,23 @@ class DummyStatefulModel {
...
+
+        ov::runtime::Tensor tensor(
+            ov::element::Type_t::f32,
+            shape,
+            values.data());
+        try{
+            inferRequest.set_tensor("input", tensor);
+        }
+        catch(const ov::Exception& e){
+            std::cout << "infer_request.set_tensor failed with message: " << e.what() << std::endl;
```
Comment (dkalinowski) [inline src/test/stateful_test_utils.hpp:158]: In OpenVINO, if we do not set the input before inference, it will be created automatically.
The test, before your changes, tested that set blob and inference work correctly.
With this change, `try catch` catches the exception and proceeds with inference, which alters the behavior of the test.
This only covers the issue we currently have?

File: src/version.hpp [hunk 2]
```diff
@@ -13,8 +13,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //*****************************************************************************
-#ifndef SRC_VERSION_HPP_
-#define SRC_VERSION_HPP_
+#ifndef MODEL_SERVER_SRC_VERSION_HPP_
```
Comment (atobiszei) [inline src/version.hpp:16]: Why this change?

File: src/test/stateful_modelinstance_test.cpp [hunk 3]
```diff
@@ -225,57 +231,35 @@ class MockedStatefulModelInstance : public ovms::StatefulModelInstance {
         sequenceManagerLock.unlock();
 
         timer.start("get infer request");
-        ovms::ExecutingStreamIdGuard executingStreamIdGuard(getInferRequestsQueue());
         ovms::ExecutingStreamIdGuard_2 executingStreamIdGuard_2(getInferRequestsQueue_2());
-        InferenceEngine::InferRequest& inferRequest = executingStreamIdGuard.getInferRequest();
         ov::runtime::InferRequest& inferRequest_2 = executingStreamIdGuard_2.getInferRequest();
         timer.stop("get infer request");
 
-        timer.start("preprocess");
```
Comment (mzegla) [inline src/test/stateful_modelinstance_test.cpp:234]: You remove timer start, but leave timer stop for preprocessing. Is that intentional?

File: src/test/stateful_modelinstance_test.cpp [hunk 4]
```diff
@@ -225,57 +231,35 @@ class MockedStatefulModelInstance : public ovms::StatefulModelInstance {
...
         if (!status.ok())
             return status;
 
         timer.start("deserialize");
-        ovms::InputSink<InferRequest&> inputSink(inferRequest);
         ovms::InputSink_2<ov::runtime::InferRequest&> inputSink_2(inferRequest_2);
         (void)inputSink_2;
         bool isPipeline = false;
-        status = ovms::deserializePredictRequest<ovms::ConcreteTensorProtoDeserializator>(*requestProto, getInputsInfo(), inputSink, isPipeline);
         timer.stop("deserialize");
```
Comment (mzegla) [inline src/test/stateful_modelinstance_test.cpp:247]: I guess timing is set in wrong places here

File: src/test/stateful_modelinstance_test.cpp [hunk 5]
```diff
@@ -225,57 +231,35 @@ class MockedStatefulModelInstance : public ovms::StatefulModelInstance {
...
-        if (!status.ok())
-            return status;
         status = ovms::deserializePredictRequest_2<ovms::ConcreteTensorProtoDeserializator_2>(*requestProto, getInputsInfo(), inputSink_2, isPipeline);
         if (!status.ok())
             return status;
 
-        timer.start("prediction");
-        status = performInference(inferRequest);
         timer.stop("prediction");
```
Comment (mzegla) [inline src/test/stateful_modelinstance_test.cpp:252]: I guess timing is set in wrong places here

File: src/test/stateful_test_utils.hpp [hunk 6]
```diff
@@ -120,6 +142,19 @@ class DummyStatefulModel {
...
+    static void setVariableState_2(ov::runtime::InferRequest& inferRequest, std::vector<float> values) {
+        DummyStatefulModel::resetVariableState_2(inferRequest);
+        std::vector<size_t> shape{1, 1};
+
+        ov::runtime::Tensor tensor(
+            ov::element::Type_t::f32,
+            shape,
+            values.data());
+
+        inferRequest.set_tensor("input", tensor);
```
Comment (dkalinowski) [inline src/test/stateful_test_utils.hpp:155]: Add comment that it throws an exception for now

Comment (dkalinowski) [reply src/test/stateful_test_utils.hpp:155]: With TODO so we can track it before release

**Review by dkalinowski — APPROVED** [review]

**Review by atobiszei — APPROVED** [review]
